### PR TITLE
Fix project edit procedure / form

### PR DESCRIPTION
**Pull request by pkitazos**
# Project Forms

- can edit project titles
- fixed bug with pre-allocated student being required even when the switch was not toggled

### CODE REVIEW

File: src/app/(protected)/[group]/[subGroup]/[instance]/projects/[id]/edit/page.tsx [hunk 1]
```diff
@@ -47,7 +47,9 @@ export default async function Page({ params }: { params: PageParams }) {
     flagTitles: project.flags.map((f) => f.title),
     capacityUpperBound: project.capacityUpperBound,
     preAllocatedStudentId: project.preAllocatedStudentId ?? "",
-    isPreAllocated: project.preAllocatedStudentId !== "",
+    isPreAllocated:
+      project.preAllocatedStudentId !== undefined &&
+      project.preAllocatedStudentId !== "",
```
Comment (JakeTrevor) [inline src/app/(protected)/[group]/[subGroup]/[instance]/projects/[id]/edit/page.tsx:52]: We should test this somehow; if we missed this edge case, are we sure we didn't miss anything else?

File: src/components/project-form.tsx [hunk 2]
```diff
@@ -103,25 +103,27 @@ export function ProjectForm({
...
 
     if (newState) {
+      // when enabling pre-allocation
       form.setValue("capacityUpperBound", 1);
+      form.setValue("isPreAllocated", true);
     } else {
+      // when disabling pre-allocation, clear the student ID
       form.setValue("preAllocatedStudentId", "");
+      form.setValue("isPreAllocated", false);
     }
```
Comment (JakeTrevor) [inline src/components/project-form.tsx:126]: Nice!

**Comment by pkitazos** [comment]
> Overall, all looks good.
> 
> Should this merge into main or staging? genuine question - at your discretion mr project manager man.

the second I opened the commit, I realised it should probably be into staging. So do i close it and open a new pr?
